### PR TITLE
Indexing ACLs - Add Kafka consumer

### PR DESCRIPTION
**Pull request by hygt**

### CODE REVIEW

File: modules/indexing/src/main/scala/ch/epfl/bluebrain/nexus/kg/indexing/acls/AclIndexer.scala [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  * Acl incremental indexing logic that pushes data into an rdf triple store.
+  *
+  * @param client    the SPARQL client to use for communicating with the rdf triple store
+  * @param contexts  the context operation bundle
+  * @param settings  the indexing settings
+  * @param F         a MonadError typeclass instance for ''F[_]''
+  * @tparam F        the monadic effect type
+  */
+@SuppressWarnings(Array("UnusedMethodParameter"))
+class AclIndexer[F[_]](client: SparqlClient[F], contexts: Contexts[F], settings: AclIndexingSettings)(
```
Comment (umbreak) [inline modules/indexing/src/main/scala/ch/epfl/bluebrain/nexus/kg/indexing/acls/AclIndexer.scala]: We don't need `contexts` here. This is just used in the other indexers to expand the context which might come in the form of a URI. In ACLs that's not used.

File: modules/indexing/src/main/scala/ch/epfl/bluebrain/nexus/kg/indexing/acls/AclIndexer.scala [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+import ch.epfl.bluebrain.nexus.commons.iam.acls.{AccessControl, Event, Path, Permission}
+import ch.epfl.bluebrain.nexus.commons.iam.identity.Identity
+import ch.epfl.bluebrain.nexus.commons.sparql.client.SparqlClient
+import ch.epfl.bluebrain.nexus.kg.core.contexts.Contexts
+
+/**
+  * Acl incremental indexing logic that pushes data into an rdf triple store.
+  *
+  * @param client    the SPARQL client to use for communicating with the rdf triple store
+  * @param contexts  the context operation bundle
```
Comment (umbreak) [inline modules/indexing/src/main/scala/ch/epfl/bluebrain/nexus/kg/indexing/acls/AclIndexer.scala]: No contexts needed

File: build.sbt [hunk 3]
```diff
@@ -1,8 +1,9 @@
-val commonsVersion     = "0.5.7"
-val metricsCoreVersion = "3.2.2"
-val jenaVersion        = "3.4.0"
-val blazegraphVersion  = "2.1.4"
-val jacksonVersion     = "2.9.0"
+val commonsVersion         = "0.5.10"
```
Comment (umbreak) [inline build.sbt]: there should be the version `0.5.11` available

File: modules/service/src/main/scala/ch/epfl/bluebrain/nexus/kg/service/queue/KafkaConsumer.scala [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+import io.circe.parser._
+import journal.Logger
+
+import scala.concurrent.{ExecutionContext, Future}
+
+object KafkaConsumer {
+
+  private val log = Logger[this.type]
+
+  final def start[T](consumerSettings: ConsumerSettings[String, String],
```
Comment (umbreak) [inline modules/service/src/main/scala/ch/epfl/bluebrain/nexus/kg/service/queue/KafkaConsumer.scala]: Missing docs

File: modules/service/src/main/scala/ch/epfl/bluebrain/nexus/kg/service/StartIndexers.scala [hunk 5]
```diff
@@ -139,6 +145,20 @@ class StartIndexers(settings: Settings, sparqlClient: SparqlClient[Future], cont
...
+
+    val aclIndexingSettings = AclIndexingSettings(settings.Sparql.Index,
+                                                  apiUri,
+                                                  settings.Sparql.Acls.GraphBaseNamespace,
+                                                  settings.Prefixes.CoreVocabulary)
+
+    val consumerSettings = ConsumerSettings(as, new StringDeserializer, new StringDeserializer)
+
+    KafkaConsumer.start[AclEvent](consumerSettings,
+                                  AclIndexer(sparqlClient, contexts, aclIndexingSettings).apply,
```
Comment (umbreak) [inline modules/service/src/main/scala/ch/epfl/bluebrain/nexus/kg/service/StartIndexers.scala]: contexts not needed here

File: modules/service/src/main/scala/ch/epfl/bluebrain/nexus/kg/service/queue/KafkaConsumer.scala [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+                     index: T => Future[Unit],
+                     topic: String)(implicit as: ActorSystem, ec: ExecutionContext, D: Decoder[T]): Future[Done] = {
+    implicit val mt: ActorMaterializer = ActorMaterializer()
+
+    Consumer
```
Comment (umbreak) [inline modules/service/src/main/scala/ch/epfl/bluebrain/nexus/kg/service/queue/KafkaConsumer.scala]: About the `offset`, what happens when we restart the service, where does this offset get stored? I don't see anywhere that we are using the `ResumableProjection` to store the offset in a Cassandra table

File: modules/service/src/test/scala/ch/epfl/bluebrain/nexus/kg/service/queue/KafkaConsumerSpec.scala [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+    """{"@context":"http://localhost/prefix/context","path":"a/b/c","identity":{"@id":"http://localhost/prefix/realms/realm/authenticated","realm":"realm","@type":"AuthenticatedRef"},"meta":{"author":{"@id":"http://localhost/prefix/realms/realm/users/alice","realm":"realm","sub":"alice","@type":"UserRef"},"instant":"1970-01-01T00:00:00.001Z"},"@type":"PermissionsRemoved"}""",
+    """{"@context":"http://localhost/prefix/context","path":"a/b/c","identity":{"@id":"http://localhost/prefix/realms/realm/users/alice","realm":"realm","sub":"alice","@type":"UserRef"},"permissions":["own","read","write"],"meta":{"author":{"@id":"http://localhost/prefix/realms/realm/users/alice","realm":"realm","sub":"alice","@type":"UserRef"},"instant":"1970-01-01T00:00:00.001Z"},"@type":"PermissionsSubtracted"}"""
+  )
+
+  "KafkaConsumer" should {
```
Comment (umbreak) [inline modules/service/src/test/scala/ch/epfl/bluebrain/nexus/kg/service/queue/KafkaConsumerSpec.scala]: We will need to add a test to check the offset, meaning: stopping the consumer and restarting it should only consume from the offset where it finished and it shouldn't re-consume events.

File: modules/service/src/main/resources/app.conf [hunk 8]
```diff
@@ -104,4 +104,11 @@ app {
     base-uri = "http://localhost:8080/v0"
     base-uri = ${?IAM_BASEURI}
   }
+
+  kafka {
+    permissions-projection-id = "permissions"
```
Comment (wwajerowicz) [inline modules/service/src/main/resources/app.conf]: I don't think this is used anywhere.

**Comment by umbreak** [comment]
Fixes #84

**Pull request merged**